examples/cors/main.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"github.com/savsgio/atreugo/v10"
+	"github.com/savsgio/atreugo/v10/middlewares"
+	"github.com/savsgio/go-logger"
+)
+
+func init() { //nolint:gochecknoinits
+	logger.SetLevel(logger.DEBUG)
+}
+
+func main() {
+	config := atreugo.Config{
+		Addr:     "0.0.0.0:8001",
+		LogLevel: "debug",
+	}
+	server := atreugo.New(config)
+
+	server.UseBefore(func(ctx *atreugo.RequestCtx) error {
```

Comment:
Middleware CORS should be executed in `UseAfter`, because `Vary` header could be overridden by other middlewares or by the view.

And the middleware could be register directly:

```go
cors := middlewares.NewCorsMiddleware(options := middlewares.CorsOptions{
	AllowedOrigins:   []string{"http://localhost:63342", "192.168.3.1:8000", "APP"},
	AllowedHeaders:   []string{"Content-Type", "content-type"},
	AllowedMethods:   []string{"GET", "POST", "OPTIONS", "PUT", "DELETE"},
	AllowCredentials: true,
	AllowMaxAge:      5600,
})

server.UseAfter(cors)
```